### ISSUE TITLE
Wait intelligently breaks after one craft

### ISSUE THREAD

**Issue by lyndistt** [Junie-774/CraftingList#12]
So after the lastest update, the wait intelligently feature works, but only while crafting one item. Any item crafted after that gives the same error as the last time. I tried with both ingame macro and plugin macro
![image](https://user-images.githubusercontent.com/123083701/215227326-f1fe0524-ec27-4292-892e-f92506410c0e.png)

I noticed that while having the wait intelligently thing on, the amount doesn't go down properly on the crafting list main menu. Below I wanted to craft 2 Basket component, but after the first one completes, the number doesn't go down from two, and the list shows missing item instead.
When I did it again with wait intelligently off, the amount goes down properly.
![image](https://user-images.githubusercontent.com/123083701/215227709-d6d1f39a-f5ae-4c88-a4be-67e3f89f71a0.png)

Here is my settings and experimental tab:
![image](https://user-images.githubusercontent.com/123083701/215227737-7b409096-185d-46d9-bc4e-36971a23b10f.png)

![image](https://user-images.githubusercontent.com/123083701/215227757-86257df3-609f-4e67-a639-7040aa383836.png)

Appreciate your effort and thank you for the help! Your plugin is awesome!

**Comment by Junie-774** [comment]
Hmmm, this is weird and concerning lol. I'm a bit busy generally with TOP prog so work on this might be a bit slow.

The confusing thing is, it seems that it's actually not receiving a response/update from the game, and is properly waiting 5s for it, after using actions.

the ingredients checklist is a known thing i'm working on, trying to find a compromise between performance and  working well, dont' wanna update the list every frame lol. You can modify the num crafts and change it back and the list will update

**Comment by Junie-774** [comment]
I can't seem to reproduce this. I'm not sure what's goin on here, as, according to the logic, it seems to be working as intended, and just isn't receiving an update from the game while crafting. Is anyone else having this issue?

**Comment by lyndistt** [comment]
I'm gonna try and recreate the issue while only running your plugin. Might be that other plugins are interfering. I'll get back to you if I discover something new 👍

**Comment by Junie-774** [comment]
Potentially! If you don't find anything new, a more complete /xllog log, starting from when the craft starts, to when it starts breaking, might give more info

**Comment by TheWalkingDude19** [comment]
I'm Having the same issue. One craft and then failed to receive a response.

**Comment by TheWalkingDude19** [comment]
I don't know what was going on last night but its working now.

**Comment by Junie-774** [comment]
hmmm. Interesting. It's definitely something with the plugin then. I wish I knew how to get more info on what's causing it ;w;

**Comment by TheWalkingDude19** [comment]
Its doing it again
![image](https://user-images.githubusercontent.com/122064331/220238323-c2562488-79df-4035-a3d0-cdf48e4bee2e.png)

**Comment by Junie-774** [comment]
hmmmm. There's a setting for how long it waits for timeout in experimental. try setting it to 10 and seeing if that does anything? It might be related with packet loss.

**Comment by lyndistt** [comment]
Hey, so I finally had some time to do some testing. I tried using the plugin by itself and it seems to have the same errors.
However, I found a fix while just randomly playing around with all the settings.
If I enable wait intelligently/ignore wait.x, then I have to disable wait after clicking synthesize and executing macro. Is this because of a user error? Maybe the wait.x feature overrides the wait feature?
![image](https://user-images.githubusercontent.com/123083701/220587866-ad6ffacc-b5fb-49bc-94e8-4bdb730414f3.png)

**Comment by Junie-774** [comment]
That's... strange. In my eternal wisdom, those checkboxes actually don't do anything other than showing the wait timers, Those checkbox options aren't actually referenced anywhere in the code other than the options tab UI.
It's not user error, those options are meant to be an additional random wait, and shouldn't have anything to do with whether macro actions respond or not.

When it fails, does the game actually perform the action? Like does groundwork go off, and the plugin just doesn't recognize it? or does the action not execute at all?

**Comment by lyndistt** [comment]
From the testing I did, no actions actually goes off because the plugin would detect an error and cancel the operation before any action input, so that's nice.

Otherwise, I think it could also be something related to connection or tick-related like you mentioned before, since making any wait durations longer in Experimental leads to less error when I tried
Here are values that doesn't cause any error:
![image](https://user-images.githubusercontent.com/123083701/220972482-ce3638ba-52cb-4e0f-9057-0b33dd77ac7c.png)
Anything less than that and it cancels the crafts after one item for me.
I tried on collectables, custom deliveries, beast tribes and regular crafts, and the problem and "fix" for me seems consistent.

**Comment by Junie-774** [comment]
Hmmm. Is the plugin actually entering the craft window while the issue persists? This is a strange error, It seems to be related to latency/packet loss, potentially while starting the craft? The plugin wouldn't be detecting any error, it just tries to use the actions. I'm not sure why the actions wouldn't be going off unless the game wasn't in the synthesis window. Though, if the synthesis window isn't available, it's strange that the plugin doesn't detect that and give an error there...

**Comment by Junie-774** [comment]
Though, to anyone experiencing this issue: try increasing the "AfterCompleteMacroHQ" options and the "AfterCompleteMacroCollectible" options under the Experimental tab

**Comment by lyndistt** [comment]
> Hmmm. Is the plugin actually entering the craft window while the issue persists? This is a strange error, It seems to be related to latency/packet loss, potentially while starting the craft? The plugin wouldn't be detecting any error, it just tries to use the actions. I'm not sure why the actions wouldn't be going off unless the game wasn't in the synthesis window. Though, if the synthesis window isn't available, it's strange that the plugin doesn't detect that and give an error there...

No, the crafting window would never show up. It would remain at the "recipe" menu

**Comment by lyndistt** [comment]
Hey, just wanted to say that I finally found the issue.

I found out that the plugin Chill Frames. Having the FPS "unlocked" while crafting and go back to locked after crafting makes the commands from Crafting list not register because the game will "lag" when the fps gets capped back down or w/e in between crafts, as the Chill Frame plugin registers it as "crafting mode" is done.

So the fix is to have it capped all the time or uncap it all the time. Any large fps drop will break it, it seems
